x/oracle/abci.go

```diff
@@ -70,26 +71,52 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) error {
 
 		//---------------------------
 		// Do miss counting & slashing
-		voteTargetsLen := len(voteTargets)
+		denomMap := map[string]map[string]bool{}
+
+		for _, denom := range voteTargets {
+			denomMap[denom] = map[string]bool{}
+		}
+
+		for denom, votes := range voteMap {
+			for _, vote := range votes {
+				// ignore denoms, not requested in voteTargets
+				_, ok := denomMap[denom]
+				if !ok {
+					continue
+				}
+
+				denomMap[denom][vote.Voter.String()] = true
+			}
+		}
 
+		// Check if each validator is missing a required denom price
 		for _, claim := range validatorClaimMap {
-			// Skip abstain & valid voters
-			if int(claim.WinCount) >= voteTargetsLen {
+
+			missing := false
+			for _, denom := range voteTargets {
+				_, ok := denomMap[denom][claim.Recipient.String()]
+				if !ok {
+					missing = true
+					break
+				}
+			}
+
+			if !missing {
```

Comment:
since this is the end of the loop it might be clearer to write `if missing { k.SetMissCounter(...) }`